fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/api/event/registry/FabricRegistryUtils.java

```diff
@@ -0,0 +1,27 @@
+package net.fabricmc.fabric.api.event.registry;
+
+import net.fabricmc.fabric.impl.registry.sync.RegistrySyncManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+public class FabricRegistryUtils {
+	/**
+	 * Register an object with a given registry that will not be synced to any clients.
+	 * If this registry entry is present on the client it will be given a new id that will not match the server!
+	 *
+	 * @param registry The target registry
+	 * @param id the {@link Identifier} to be used to register the entry
+	 * @param entry the registry entry to be added to the target registry
+	 * @param <T> The type of the registry/entry
+	 * @return Returns the input entry
+	 */
+	public static <T> T registerServerOnlyRegistryEntry(Registry<T> registry, Identifier id, T entry) {
+		if (!RegistryAttributeHolder.get(registry).hasAttribute(RegistryAttribute.SYNCED)) {
```

Comment:
This may cause an issue if the registry is untouched before you try to add a server only entry. As adding any new entries will mark the registry as synced.

Reply:
Is fabric adding SYNCED flags this way? weird. Also worth discussing is whether the mark for server only and the registration should be bound together or separate.

Reply:
Not nearly that simple, but yes. Other registries are marked as synced (which well requires an entry per registry) since some registries refer to entries in packets by raw id

Reply:
Hmm, Registry's codec functionality can store elements as int ids if it uses compression (which afaik minecraft doesn't yet use).